packages/frontend/helpers/balances.ts

```diff
@@ -0,0 +1,132 @@
+import { Address, ChainId, Token } from '@x-fuji/sdk';
+import { formatUnits } from 'ethers/lib/utils';
+
+import { BALANCE_POLLING_INTERVAL } from '../constants';
+import { sdk } from '../services/sdk';
+import { onboard, useAuth } from '../store/auth.store';
+import { useBorrow } from '../store/borrow.store';
+
+export const fetchBalances = async (
+  tokens: Token[],
+  address: string,
+  chainId: ChainId
+) => {
+  const rawBalances = await sdk.getTokenBalancesFor(
+    tokens,
+    Address.from(address),
+    chainId
+  );
+  const balances: Record<string, number> = {};
+  rawBalances.forEach((b, i) => {
+    const value = parseFloat(formatUnits(b, tokens[i].decimals));
+    balances[tokens[i].symbol] = value;
+  });
+  return balances;
+};
+
+export const updateNativeBalance = (addr?: string) => {
+  const address = addr ?? useAuth.getState().address;
+  if (!address) return;
+  onboard.state.actions.updateBalances([address]);
+};
+
+export const checkBalances = async () => {
+  const address = useAuth.getState().address;
+  if (!address) return;
+
+  // Triggers a native token refresh using onboard
+  updateNativeBalance(address);
+
+  if (!shouldFetchERC20) return;
+
+  // Now let's to with ERC20 tokens
+  const debt = useBorrow.getState().debt;
+  const collateral = useBorrow.getState().collateral;
+
+  const data = [collateral, debt];
+
+  data.forEach(async (asset, i) => {
```

Comment:
I'd suggest creating a new function `checkBalance(type)` that will be called from the current `checkBalances()` so that we'll avoid the forEach for just two elements and referencing "types" by index.